scrapper.py

```diff
@@ -26,38 +36,123 @@ class Crawler:
+            self.urls.append('https://www.m24.ru' + link['href'])
 
     def find_articles(self):
         """
         Finds articles
         """
-        pass
+        software_names = [SoftwareName.CHROME.value]
+        operating_systems = [OperatingSystem.WINDOWS.value, OperatingSystem.LINUX.value]
+        useragent_random = UserAgent(software_names=software_names, operating_systems=operating_systems, limit=100)
+        headers = {'user-agent': useragent_random.get_random_user_agent(),
+                   'accept': '*/*', 'accept-language': 'ru-RU,ru;q=0.9,en-US;q=0.8,en;q=0.7'}
+
+        for seed_url in self.seed_urls:
+            response = requests.get(seed_url, headers=headers, allow_redirects=True, timeout=5)
+            print(response.status_code)
+            with open('index1.html', 'w', encoding='utf-8') as file:
+                file.write(response.text)
+            print(response.status_code)
+            article_bs = BeautifulSoup(response.text, 'html.parser')
+            self._extract_url(article_bs)
 
     def get_search_urls(self):
         """
         Returns seed_urls param
         """
-        pass
+        return self.seed_urls
+
+
+class HTMLParser:
+    def __init__(self, article_url, article_id):
+        self.article_url = article_url
+        self.article_id = article_id
+        self.article = Article(article_url, article_id)
+
+    def parse(self):
+        response = requests.get(self.article_url)
+        # прописать обманку для сайта!!!
+        article_bs = BeautifulSoup(response.text, 'html.parser')
+
+        self._fill_article_with_text(article_bs)
+        self._fill_article_with_meta_information(article_bs)
+        self.article.save_raw()
+
+        return self.article
+
+    def _fill_article_with_text(self, article_bs):
+
+    def _fill_article_with_meta_information(self, article_bs):
 
 
 def prepare_environment(base_path):
     """
     Creates ASSETS_PATH folder if not created and removes existing folder
     """
-    pass
+    try:
+        os.rmdir(base_path)
+    except FileNotFoundError:
+        pass
+    os.mkdir(base_path)
 
 
 def validate_config(crawler_path):
     """
     Validates given config
     """
-    pass
+    with open(crawler_path, 'r', encoding='utf-8') as file:
+        config = json.load(file)
+        seed_urls = config['seed_urls']
+        max_articles = config['total_articles_to_find_and_parse']
+
+        if not isinstance(seed_urls, list):
+            raise IncorrectURLError
+        # если переменная seed_urls не является списком
+
+        if not seed_urls:
+            raise IncorrectURLError
+        # возбуждает исключение, если такого seed_url
+        # нет в списке, который мы сделали в scrapper_config.json
+
+        if not isinstance(max_articles, int):
+            raise IncorrectNumberOfArticlesError
+        # если число статей, которое мы
+        # передаем в scrapper_config.json не является integer
+
+        if max_articles > 100:
```

Comment:
I recommend taking a bigger number as 100 is a required number of articles to find and parse, so it is a bit odd that this code is expected to work for 100 items but is going to complain about 101

Reply:
okay, i'll take a bigger number of articles